src/aioamazondevices/api.py

```diff
@@ -598,127 +579,83 @@ async def _register_device(
-            input_data=_data,
-            json_data=True,
-        )
-        json_data = await raw_resp.json()
-
-        final_sensors: dict[str, dict[str, AmazonDeviceSensor]] = {}
-        for sensors in json_data["deviceStates"]:
-            _id = sensors["entity"]["entityId"]
-            dict_sensors: dict[str, AmazonDeviceSensor] = {}
-            for sensor in sensors["capabilityStates"]:
-                sensor_json = orjson.loads(sensor)
-                if sensor_json["name"] in SENSORS:
-                    _value = sensor_json["value"]
-                    _value_dict = isinstance(_value, dict)
-                    _name = sensor_json["name"]
-                    dict_sensors.update(
-                        {
-                            _name: AmazonDeviceSensor(
-                                name=_name,
-                                value=(_value["value"] if _value_dict else _value),
-                                scale=_value.get("scale") if _value_dict else None,
-                            )
-                        }
+        devices_state = await self._get_devices_state()
+        devices_sensors: dict[str, dict[str, AmazonDeviceSensor]] = {}
+        devices_endpoints: dict[str, dict[str, Any]] = {}
+
+        endpoints = devices_state["data"]["listEndpoints"]
+        for endpoint in endpoints["endpoints"]:
+            serial_number = (
+                endpoint["serialNumber"]["value"]["text"]
+                if endpoint["serialNumber"]
+                else None
+            )
+            if serial_number in self._devices:
+                devices_sensors[serial_number] = self._get_device_sensor_state(endpoint)
+                devices_endpoints[serial_number] = endpoint
+
+        return devices_endpoints, devices_sensors
+
+    def _get_device_sensor_state(
+        self, endpoint: dict[str, Any]
+    ) -> dict[str, AmazonDeviceSensor]:
+        device_sensors: dict[str, AmazonDeviceSensor] = {}
+        if endpoint_dnd := endpoint.get("settings", {}).get("doNotDisturb"):
+            device_sensors["dnd"] = AmazonDeviceSensor(
+                "dnd", endpoint_dnd.get("toggleValue"), None
+            )
+        if endpoint["features"]:
+            for feature in endpoint.get("features", {}):
+                first_property = feature["properties"][0]
+                if feature["name"] == "temperatureSensor":
+                    device_sensors["temperature"] = AmazonDeviceSensor(
+                        "temperature",
+                        first_property["value"]["value"],
+                        first_property["value"]["scale"],
+                    )
+                if feature["name"] == "motionSensor":
+                    device_sensors["humanPresenceDetectionState"] = AmazonDeviceSensor(
+                        "humanPresenceDetectionState",  # name to match legacy value
+                        first_property.get("detectionStateValue", "NOT_SET")
```

Comment:
We may prefer to report the raw value here and manipulate it on HomeAssistant side